practice-app/website/api/event.py

```diff
@@ -76,5 +79,48 @@ def event():
         except exc.SQLAlchemyError as e:
             db.session.rollback()
             return "Try Later", 403
-        
+
         return jsonify(new_event.serialize()), 201
+
+
+@events.route('<event_id>/discussions', methods=['GET', 'POST'])
+@swag_from('doc/discussions_GET.yml', methods=['GET'])
+@swag_from('doc/discussions_POST.yml', methods=['POST'])
+def discussionForEvent(event_id):
+
+    if request.method == 'POST':
+
+        if event_id < 0:
+            return "Wrong parameters", 401
+
+        req = requests.get(
+            'https://api.namefake.com/english-united-states/random/')
+        data = json.loads(req.content)
+        name_shown = data['name']
+
+        message = request.json['text']
+        if len(message) < 1:
+            return "Text field cannot be empty.", 402
+
+        message = name_shown + ': ' + message
+
+        discussionPostList = DiscussionPost.query.all()
+
+        doesExist = False
+        for i in range(len(discussionPostList)):
+            if discussionPostList[i].serialize()["id"] == int(event_id):
+                # exists
+                doesExist = True
+
+        if not doesExist:
+            newPost = DiscussionPost(id=event_id, text=message)
```

Comment:
The check for invalid event_id is not made. Users should not be able to add a discussion for a non-existent event.

Reply:
I agree, you can try to get an event from the db by id and check if returns none

Reply:
Now I added this check, thank you.